.github/workflows/build-dotnet.yml

```diff
@@ -96,6 +103,8 @@ jobs:
 
     - name: Checkout
       uses: Lombiq/GitHub-Actions/.github/actions/checkout@dev
+      with:
+        token: ${{ env.CHECKOUT_TOKEN }}
```

Comment:
This will be empty by default, and thus no authentication token will be passed to `actions/checkout` (overwriting the default `github.token`). This will fail checkout if the current repository is authenticated. Do what e.g. `build-and-test-orchard-core` does, which allows using `github.token` if no other token is specified (which is the desired case if there are no private submodules).

Perhaps that logic can be somewhat DRYd, but I'm not sure how much, since composite actions need secrets passed in explicitly, they can't access them from the context.

Unless `actions/checkout` takes an empty string parameter as no parameter and will use `github.token` instead; then the default-handling logic in other workflows is unnecessary, so please remove that.

Reply:
Gotcha, I'll test in NEST then like this and we'll see which way to change it!

Reply:
We'll probably need to use that logic, because 3 out of 4 other workflows that already use the secret has it. The 4th one is test-analysis-failure, but it's not used in a situation yet where the token is needed, that's why it didn't come up there. I'll update that too.

BTW the logic can't be factored out to our checkout action?

Reply:
I'm not entirely sure if the default-handling logic is actually needed, so please do check.

You can actually check for an empty `token` input in our `checkout` action and use `github.token` (or `$Env:GITHUB_TOKEN`) if it's empty, because I think the GitHub token, unlike any secret, is actually available in composite actions too. And then you can just directly pass `CHECKOUT_TOKEN` to `token` in all consumers without a separate "Set Checkout Token" step.

Reply:
No, `$Env:GITHUB_TOKEN` is not available in actions by default, needs to be explicitly passed in, see e.g. `post-pull-request-checks-automation`.

Reply:
Yeah, my thoughts exactly - copy-paste the same step and just shuffle around some of the variables.
It seems to work, I just made a mistake in the deployment workflow at first, otherwise it will be good to go and all the workflows calling the checkout action can be simplified.